### PR TITLE
feat: uab package list model and proxy model

### PR DESCRIPTION
**Pull request by rb-union**
feat: uab package list model

Adds uab package list model.
Manages the backend of the add/remove and
install/uninstall processes for uab packages.

Log: Add uab package list model.
Influence: uab-package

feat: mixed deb/uab pacakge installation.

Add proxy package list model.
* Mixed deb/uab package installation flow;
* Connect DebInstaller and backend model signals,
  interfaces, etc.

Log: Mixed uab/deb pacakge installation.
Influence: uab-package

feat: commit changes to uab backend

Commit changes to uab backend after install/uninstall.

Log: Commit changes to uab backend.
Influence: uab-package

### CODE REVIEW

File: src/deb-installer/manager/packagesmanager.cpp [hunk 1]
```diff
@@ -691,8 +692,9 @@ void PackagesManager::slotDealDependResult(int iAuthRes, int iIndex, const QStri
             getPackageDependsStatus(iIndex);
             if (!m_dependsPackages.isEmpty()) {
                 qInfo() << m_dependsPackages.size() << m_dependsPackages.value(m_currentPkgMd5).second.size();
-                if (1 == m_preparedPackages.size()) {}
-                    // Q_EMIT signalDependPackages(m_dependsPackages.value(m_currentPkgMd5), false);
+                if (1 == m_preparedPackages.size()) {
```
Comment (pengfeixx) [inline src/deb-installer/manager/packagesmanager.cpp:695]: 判断内缺少内容

**Review by starhcq — APPROVED** [review]

**Comment by pengfeixx** [comment]
[是否满足兼容性要求] Y
[是否满足commit提交规范] Y
[是否满足编码规范] Y
[Review结论] Pass
[Fail原因] N/A

File: src/deb-installer/utils/utils.cpp [hunk 2]
```diff
@@ -287,53 +287,69 @@ QString Utils::holdTextInRect(const QFont &font, const QString &srcText, const i
...
 }
 
 int Utils::compareVersion(const QString &v1, const QString &v2)
 {
     return QApt::Package::compareVersion(v1, v2);
 }
 
+Pkg::PackageType Utils::detectPackage(const QString &filePath)
+{
+    const QMimeDatabase db;
```
Comment (starhcq) [inline src/deb-installer/utils/utils.cpp:340]: 没有必要const

File: src/deb-installer/model/proxy_package_list_model.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+private:
+    void nextModelInstall();
+
+    ModelPtr modelFromType(Pkg::PackageType type);
+    ModelPtr addModel(Pkg::PackageType type);
+    void connectModel(ModelPtr model);
+
+    QPair<ModelPtr, int> findFromProxyIndex(int proxyIndex) const;
+    int proxyIndexFormModel(ModelPtr findModel, int index);
```
Comment (starhcq) [inline src/deb-installer/model/proxy_package_list_model.h:40]: 接口命名错误

File: src/deb-installer/utils/utils.cpp (hunk 2 above)
Comment (rb-union) [reply src/deb-installer/utils/utils.cpp:340]: 已更新 https://github.com/linuxdeepin/deepin-deb-installer/pull/263

File: src/deb-installer/model/proxy_package_list_model.h (hunk 3 above)
Comment (rb-union) [reply src/deb-installer/model/proxy_package_list_model.h:40]: 已更新 https://github.com/linuxdeepin/deepin-deb-installer/pull/263

**Pull request merged**